extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -236,12 +236,26 @@ class LifecycleTask extends BackbeatTask {
                 }
             } else {
                 // 2. by VersionId, lower number means newer
+                //    NOTE: VersionId may be null
+                const nullVersion = (versions[vIdx].VersionId === 'null'
+                    || deleteMarkers[dmIdx].VersionId === 'null');
                 const isVersionVidNewer = (versions[vIdx].VersionId <
                     deleteMarkers[dmIdx].VersionId);
-                if (isVersionVidNewer) {
+                if (!nullVersion && isVersionVidNewer) {
```

Comment:
You can actually group the null version logic in a `if (nullVersion) {}` block, as Rahul suggested, even though there will be an extra nesting level of `if` it makes it easier to read and understand overall.